src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/ConfigCommands/ConfigRunners.cs

```diff
@@ -74,6 +74,23 @@ public static void Run(ConfigGetArgs args, Func<ILogger> getLogger)
         }
     }
 
+    internal static class ConfigUnsetRunner
+    {
+        public static void Run(ConfigUnsetArgs args, Func<ILogger> getLogger)
+        {
+            RunnerHelper.EnsureArgumentsNotNull(args, getLogger);
+            RunnerHelper.ValidateConfigKey(args.ConfigKey);
+            ISettings settings = string.IsNullOrEmpty(args.ConfigFile)
+                ? RunnerHelper.GetSettingsFromDirectory(null)
+                : RunnerHelper.GetSettingsFromFile(args.ConfigFile);
+
+            if (!SettingsUtility.DeleteConfigValue(settings, args.ConfigKey))
+            {
+                throw new CommandException(string.Format(CultureInfo.CurrentCulture, Strings.Error_ConfigUnsetNonExistingKey, args.ConfigKey));
```

Comment:
I think the "key does not exist" scenario might warrant some customer development. If a key is not already set, then the nuget.config files are already in the customer's desired state. If a customer is running this is a CI script, do we want to fail their build because their machine state is already in the desired state?

Also, consider `git config --unset user.invalid`. It doesn't output any text, but it does set a non-zero exit code. So, perhaps this is precedent that we return non-zero exit codes.

.NET APIs like `List<T>.Remove`, `Dictionary<TKey, TValue>Remove`, they don't throw when the item isn't found, though they do have a `bool` return value saying whether anything was actually removed.

We could have a comment line option to allow customers to opt-in/out of error scenarios. If we default to "zero exit code", then I'd suggest `--strict` as the option to error when the key doesn't exist. If we default to strict mode, I don't have a good suggestion for an `--arg` that customers could use in scripts to avoid causing the script to error.

Reply:
Got it! The spec mentions that a message should be displayed to the user if the key could not be unset, so I'll change this to just log the message instead of throwing an exception.